File: backend/src/main/scala/microservices/traveler/objects/TravelerSpecialRequirementInfo.scala
// 本文件定义 traveler 模块的特殊需求传输对象 `TravelerSpecialRequirementInfo`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerSpecialRequirementInfo(
    assistanceType: String,
    requirementNote: Option[String],
    hasLargeLuggage: Boolean,
    luggageNote: Option[String]
)

object TravelerSpecialRequirementInfo:
  given sourceEncoder: Encoder[TravelerSpecialRequirementInfo] = deriveEncoder
  given sourceDecoder: Decoder[TravelerSpecialRequirementInfo] = deriveDecoder

File: frontend/src/microservices/traveler/objects/TravelerSpecialRequirementInfo.ts
// 本文件定义 traveler 模块的 `TravelerSpecialRequirementInfo`，作为特殊需求传输数据并提供 JSON 编解码。

export type TravelerSpecialRequirementInfo = {
  assistanceType: string
  requirementNote: string | null
  hasLargeLuggage: boolean
  luggageNote: string | null
}

export const travelerSpecialRequirementInfoFromJson = (json: string): TravelerSpecialRequirementInfo =>
  JSON.parse(json) as TravelerSpecialRequirementInfo

export const travelerSpecialRequirementInfoToJson = (value: TravelerSpecialRequirementInfo): string =>
  JSON.stringify(value)
